src/main/java/org/acra/collector/Collector.java

```diff
@@ -0,0 +1,66 @@
+package org.acra.collector;
+
+import android.support.annotation.NonNull;
+
+import org.acra.ReportField;
+import org.acra.builder.ReportBuilder;
+
+import java.util.Set;
+
+/**
+ * Created by Lukas on 11.08.2016.
+ */
+public abstract class Collector implements Comparable<Collector> {
+    private final ReportField[] reportFields;
+
+    protected Collector(ReportField... reportFields) {
+        this.reportFields = reportFields;
+    }
+
+    /**
+     * @return all fields this collector can collect
+     */
+    @NonNull
+    public final ReportField[] canCollect() {
+        return reportFields;
+    }
+
+    /**
+     * Should be:
+     * > 0 if this should be called as early as possible
+     * = 0 if the relative position isn't important
+     * < 0 if this is dodgy
+     *
+     * @return priority of this collector
+     */
+    public int getPriority() {
```

Comment:
That was meant to provide control over the execution order to the collectors, but it is probably overengineering to do so.